### PR TITLE
Introduces a plan.proto for Android-friendly serialization of a Plan.

### PR DESCRIPTION
**Pull request by piotrswigon**
Replaces manifest.proto with plan.proto which provides a reasonable starting point for plan to proto serialization.

The serializer is not yet updated to fill in all the details, it will come in the next PR.

The controversial bit worth discussing is that I re-used the main type hierarchy for schema field types, instead of having a dedicated parallel type hierarchy. The reason being that schema fields can now be references and collections, which are concepts already represented in the main type hierarchy. It is however true that this proto can now be used to represent types that we cannot operate on, e.g. a handle of a primitive type. I think that's reasonable, but I don't feel all that strongly.

### CODE REVIEW

File: java/arcs/core/data/plan.proto [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+option java_multiple_files = true;
+
+// Holds a single plan and particle specs used by it.
+message PlanEnvelope {
+  Plan plan = 1;
+  repeated ParticleSpec particle_specs = 2;
+}
+
+// Plan: A resolved recipe that can be instantiated.
+message Plan {
```
Comment (csilvestrini) [inline java/arcs/core/data/plan.proto]: FYI I opted to use "Proto" as a suffix to help disambiguate between the main class and the proto (VersionMap vs. VersionMapProto). I know you have this in a separate package, which means the symbols won't clash, but it does mean if you want to refer to both in the same file you will have to fully qualify one of them, which is annoying.

For anything that has a one-to-one correspondence with a Kotlin class (e.g. `Plan`), I'd recommend using the suffix (`PlanProto`).

Comment (piotrswigon) [reply java/arcs/core/data/plan.proto]: Good plan. There definitely will be places (like serialization and deserialization) where we will want to import both. Added Proto to all messages here.

File: javatests/arcs/core/data/BUILD [hunk 2]
```diff
@@ -13,7 +13,7 @@ arcs_kt_jvm_test_suite(
     package = "arcs.core.data",
```
Comment (alxrsngrtn) [inline javatests/arcs/core/data/BUILD]: You need to add the *Test.kt files as sources.

File: java/arcs/core/data/recipe.proto [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+}
+
+message TypeVariableProto {
+  // Identifies a type variable in a recipe or a particle spec.
+  // Type variables with the same name are linked together.
+  string name = 1;
+} 
+
+message SchemaProto {
+  repeated string names = 1;
```
Comment (alxrsngrtn) [inline java/arcs/core/data/recipe.proto]: Schema should also have a hash.

File: javatests/arcs/core/data/BUILD (hunk 2 above)
Comment (piotrswigon) [reply javatests/arcs/core/data/BUILD]: arcs_kt_jvm_test_suite does that as a default.

**Pull request merged**